index.js

```diff
@@ -29,21 +29,45 @@ class LogStreamingPlugin {
 	}
 
 	logServerless() {
+		let logHandlerLambdaName;
+		const logHandlerFnName = _.get(this.serverless, "service.custom.logHandler.function", "loghandler");
 		_.forEach(this.serverless.service.getAllFunctions(), functionName => {
-			if (functionName === "loghandler") {
+			const functionLogicalId = `${_.upperFirst(functionName)}LambdaFunction`;
+			if (functionName === logHandlerFnName) {
+				// Do not add the loghandler to itself.
+				logHandlerLambdaName = functionLogicalId;
 				return;
 			}
-			const functionLogicalId = `${_.upperFirst(functionName)}LambdaFunction`;
 			const functionObject = this.serverless.service.getFunction(functionName);
-
+			if (_.get(functionObject, "loghandler", true) === false) {
+				// If the loghandler property is set to false then do not create a loghandler
+				return;
+			}
 			cloudwatchLogsSubscriptionFilterTemplate.Properties.LogGroupName = `/aws/lambda/${functionObject.name}`;
 			const newResources = {
 				[`${functionLogicalId}SubscriptionFilter`]: cloudwatchLogsSubscriptionFilterTemplate
 			};
 
 			_.merge(this.serverless.service.provider.compiledCloudFormationTemplate.Resources, newResources);
-			
 		});
+		
+		if (!_.has(this.serverless.service.provider.compiledCloudFormationTemplate.Resources, "LoggingLambdaPermission")) {
```

Comment:
I'm not seeing anything in the function definitions that fits what we're looking for. I'm hoping since we define the resource by name within the plugin that it's okay to reference it by that string.